### PR TITLE
[FIX] ctc_loss_op_test_gpu unit test suite failure

### PR DESCRIPTION
**Pull request by arsic3**
## Motivation

The command  ``` bazel test -s --config=rocm //tensorflow/python/kernel_tests/nn_ops:ctc_loss_op_test_gpu ``` will produce failure of CTCLossDeterministicTest.testForwardAndBackward on gfx11xx and gfx12xx. The failure occured due to one element that represents the difference between gradients is greater than the previous absolute tolerance.

Log:
```
======================================================================
FAIL: testForwardAndBackward2 (True, False) (__main__.CTCLossDeterministicTest) [graph_mode]
CTCLossDeterministicTest.testForwardAndBackward2 (True, False)
testForwardAndBackward(True, False)
----------------------------------------------------------------------
...
Mismatched elements: 1 / 32000 (0.00313%)
Max absolute difference among violations: 7.075071e-05
Max relative difference among violations: 0.00036804
...
======================================================================
FAIL: testForwardAndBackward3 (True, True) (__main__.CTCLossDeterministicTest) [graph_mode]
CTCLossDeterministicTest.testForwardAndBackward3 (True, True)
testForwardAndBackward(True, True)
----------------------------------------------------------------------
...
Mismatched elements: 1 / 32000 (0.00313%)
Max absolute difference among violations: 5.2332878e-05
Max relative difference among violations: 0.00028655
...
----------------------------------------------------------------------
```
## Technical Details

Increase the absolute tolerance between gradients in CTCLossDeterministicTest.testForwardAndBackward due to the floating-point arithmetic.

### CODE REVIEW

**Review by i-chaochen — COMMENTED** [review]
I suppose this should only for navi?

File: tensorflow/python/kernel_tests/nn_ops/ctc_loss_op_test.py [hunk 1]
```diff
@@ -1234,7 +1234,9 @@ def testForwardAndBackward(self, sparse_labels, logits_time_major):
             (loss_a, loss_b, gradient_a, gradient_b))
         self.assertAllEqual(loss_a, loss_b, "Loss mismatch")
         # self.assertAllEqual(gradient_a, gradient_b, "Gradient mismatch")
-        self.assertAllClose(gradient_a, gradient_b, atol=5e-05)
+        # adding absolute tolerance to the difference of gradients due to the 
+        # floating point arithmetics
+        self.assertAllClose(gradient_a, gradient_b, atol=1e-04)
```
Comment (i-chaochen) [inline tensorflow/python/kernel_tests/nn_ops/ctc_loss_op_test.py]: I suppose this should only for navi?

**Review by ScXfjiang — APPROVED** [review]

File: tensorflow/core/common_runtime/gpu/gpu_device.cc [hunk 2]
```diff
@@ -1970,6 +1970,7 @@ static string GetShortDeviceDescription(
 #elif TENSORFLOW_USE_ROCM
   return strings::StrCat("device: ", platform_device_id.value(),
                          ", name: ", desc.name(),
+                         ", gcn arch: ", desc.rocm_compute_capability().gcn_arch_name(),
```
Comment (ScXfjiang) [inline tensorflow/core/common_runtime/gpu/gpu_device.cc:1973]: Would it be better if we use the same key-value pairs as CUDA, though they return the same thing?

`compute capability: desc.rocm_compute_capability().ToString()`

File: tensorflow/python/framework/test_util.py [hunk 3]
```diff
@@ -189,6 +189,29 @@ def gpu_device_name() -> str:
       return compat.as_str(x.name)
   return ""
 
+@tf_export("test.gpu_gcn_arch")
```
Comment (ScXfjiang) [inline tensorflow/python/framework/test_util.py:192]: Does CUDA have the counterpart? What's the behavior if this API is called on the CUDA platform?

Should we add the `rocm_` prefix here?

Comment (arsic3) [reply tensorflow/python/framework/test_util.py:192]: If this method is called on a system with CUDA support, it will just return empty string, as the CUDA description is not the same as the the ROCm one.

We could be more descriptive with the name and detail design comment.

File: tensorflow/core/common_runtime/gpu/gpu_device.cc (hunk 2 above)
Comment (arsic3) [reply tensorflow/core/common_runtime/gpu/gpu_device.cc:1973]: Here we return a specific part of the compute capability; therefore I've used getter specific for returning the gcn_arch_name. 
Otherwise, if we inspect [device_description.h](https://github.com/ROCm/tensorflow-upstream/blob/574fddb8a0db90b99ea68856641f237de251ee22/third_party/xla/xla/stream_executor/device_description.h) we can observe that CudaComputeCapability has method ToString() that returns the version of the software in a format "major"."minor".  Inspecting RocmComputeCapability we can see that there is no equivalent method in it. Nevertheless, it would not fit our needs. :)

**Review by i-chaochen — APPROVED** [review]
LGTM
